app/org/sagebionetworks/bridge/dynamodb/DynamoSurveyResponseDao.java

```diff
@@ -59,25 +49,33 @@ public void setSurveyDao(DynamoSurveyDao surveyDao) {
         this.surveyDao = surveyDao;
     }
     
-    @Override
-    public SurveyResponse createSurveyResponse(GuidCreatedOnVersionHolder keys, String healthCode,
-            List<SurveyAnswer> answers) {
-        return createSurveyResponseInternal(keys, healthCode, answers, BridgeUtils.generateGuid());
+    @Autowired
+    public void setDistributedLockDao(DistributedLockDao lockDao) {
+        this.lockDao = lockDao;
     }
-
+    
     @Override
     public SurveyResponse createSurveyResponse(GuidCreatedOnVersionHolder keys, String healthCode,
             List<SurveyAnswer> answers, String identifier) {
+        
+        String key = RedisKey.SURVEY_RESPONSE_LOCK.getRedisKey(identifier);
+        String lock = null;
         try {
+            lock = lockDao.acquireLock(SurveyResponse.class, key);
             SurveyResponse response = getSurveyResponseInternal(healthCode, identifier);
             if (response == null) {
-                // This is the response we want. This is good, carry on.
                 return createSurveyResponseInternal(keys, healthCode, answers, identifier);
             }
```

Comment:
We actually don't even need get-before-put semantics, since DDB optimistic locking also takes care of this for us.